llvm/utils/TableGen/GlobalISel/Patterns.cpp

```diff
@@ -0,0 +1,845 @@
+//===- Patterns.cpp --------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//
+//===----------------------------------------------------------------------===//
+
+#include "Patterns.h"
+#include "../CodeGenInstruction.h"
```

Comment:
Probably shouldn't use relative include paths

Reply:
Unfortunately needed, linkage of that library is a bit of a mess, I'll create a patch to clean it up after this.